### PR TITLE
LPS-179782 search-experiences-web: Rerender asset types modal to get most current list

### PR DESCRIPTION
**Pull request by oliv-yu**
https://issues.liferay.com/browse/LPS-179782 - Blueprints deselect asset type doesn't remove from pop-up

It seemed like the issue was that the `modalSelectedTypes` state inside `SearchableTypesModal` wouldn't be the most up-to-date since the user is able to delete asset types on the main page, but the modal doesn't update for that. Luckily, the outdated list doesn't get saved with the blueprint. To address the bug, the component `SearchableTypesModal` just gets rendered again with the most current `selectedTypes`.

Let me know what you think, several other search experiences modals get rendered again when visibility is set to true (like `EditTitleModal`). I don't think I noticed other cases of this bug happening, but if you do, lmk!

### CODE REVIEW

**Comment by oliv-yu** [comment]
ci:test:sf

**Comment by oliv-yu** [comment]
ci:test:search

File: modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/edit_sxp_blueprint/query_builder_tab/SearchableTypesModal.js [hunk 1]
```diff
@@ -56,177 +52,189 @@ function SearchableTypesModal({
...
+	onFetchSearchableTypes,
+	onFrameworkConfigChange,
+	searchableTypes,
+	selectedTypes,
+}) {
+	const [visible, setVisible] = useState(false);
+
+	const {observer, onClose} = useModal({
+		onClose: () => setVisible(false),
+	});
```
Comment (thektan) [inline modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/edit_sxp_blueprint/query_builder_tab/SearchableTypesModal.js:222]: Think you can reduce this to:

```javascript
const {observer, onOpenChange, open} = useModal();
```

See https://clayui.com/docs/components/modal.html

**Review by thektan — CHANGES_REQUESTED** [review]
The issue is that initial state is ignored after the first render, so using props as initial state is an anti-pattern. This creates multiple sources-of-truth as well. React docs have a blurb about this: https://react.dev/learn/choosing-the-state-structure#don-t-mirror-props-in-state

I'd recommend renaming like the docs say with `initial*` and adding a `useEffect` to make sure the `modalSelectedTypes` is re-calculated whenever the `initial*` value changes. 

Although can you also do a bit of research if there's a better practice than using `useEffect` to keep the `modalSelectedTypes` in sync with the initial value?

**Comment by oliv-yu** [comment]
@thektan Added the change to adjust the state upon render, and tested it out. Let me know if that works? Thanks!

**Review by thektan — APPROVED** [review]
Thanks for making those changes!

**Comment by thektan** [comment]
ci:forward

**Comment by thektan** [comment]
None of the failures seem related.

ci:forward:force

**Comment by thektan** [comment]
ci:forward:force

**Pull request closed without merging**